### PR TITLE
Switch to faraday from rest-client

### PR DESCRIPTION
**Pull request by oldmax**
We were originally going to use the Cloudinary gem in our rails application, but the dependency on RestClient is a bit of a deal breaker. We would much rather be using Faraday for a few reasons.
1. Significantly better syntax, [see documentation here](http://www.intridea.com/blog/2012/3/12/faraday-one-http-client-to-rule-them-all).
2. Proxy settings in RestClient are stored in global state, so can't be set per-request
3. [SSLv3 issues](https://www.google.com/webhp?q=restclient+sslv3+alert+handshake+failure#q=restclient+sslv3+alert+handshake+failure)

(Sorry for the messy diffs -- looks like my text editor decided to remove trailing whitespace)

### CODE REVIEW

File: cloudinary.gemspec [hunk 1]
```diff
@@ -20,7 +20,7 @@ Gem::Specification.new do |s|
   s.executables   = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
   s.require_paths = ["lib"]
 
-  s.add_dependency "rest-client"
+  s.add_dependency "faraday"
```
Comment (jimpo) [inline cloudinary.gemspec]: You should specify a version.

**Comment by thedrow** [comment]
Any chance to see this merged?

**Comment by frsantos** [comment]
Just passing by to bump this up.

`rest-client` seems to be abandoned, with [just a single release in 2019](https://www.ruby-toolbox.com/projects/rest-client) after no releases since 2017.

Rest-client seems one of the main gems still using `mime-types` gem, as the rest have migrated to the less memory hungry `mini_mimes`. Cloudinary is one of two gems we still use having a dependency on `mime-types` and we'd like to remove it.

**Comment by wissam-khalili** [comment]
Hi @frsantos ,

Thank you for your input.
I have asked our team to check it and I will keep you posted.

Regards,
Wissam

**Pull request closed without merging**